python/ray/autoscaler/_private/commands.py

```diff
@@ -973,47 +979,53 @@ def rsync(config_file: str,
                 break
 
     provider = _get_node_provider(config["provider"], config["cluster_name"])
+
+    def rsync_to_node(node_id, is_head_node):
+        updater = NodeUpdaterThread(
+            node_id=node_id,
+            provider_config=config["provider"],
+            provider=provider,
+            auth_config=config["auth"],
+            cluster_name=config["cluster_name"],
+            file_mounts=config["file_mounts"],
+            initialization_commands=[],
+            setup_commands=[],
+            ray_start_commands=[],
+            runtime_hash="",
+            use_internal_ip=use_internal_ip,
+            process_runner=_runner,
+            file_mounts_contents_hash="",
+            is_head_node=is_head_node,
+            docker_config=config.get("docker"))
+        if down:
+            rsync = updater.rsync_down
+        else:
+            rsync = updater.rsync_up
+
+        if source and target:
+            # print rsync progress for single file rsync
+            cmd_output_util.set_output_redirected(False)
+            set_rsync_silent(False)
+            rsync(source, target, is_file_mount)
+        else:
+            updater.sync_file_mounts(rsync)
+
     try:
         nodes = []
-        if all_nodes:
-            # technically we re-open the provider for no reason
-            # in get_worker_nodes but it's cleaner this way
-            # and _get_head_node does this too
-            nodes = _get_worker_nodes(config, override_cluster_name)
-
         head_node = _get_head_node(
```

Comment:
Can we move this into the `else` since it's potentially super expensive/could have side effects if Tune uses rsync without the autoscaler?

Reply:
I originally had it there, but you want to do special processing for the head node case (required for docker to do some other processing iir). 

Tune won't use this command without the autoscaler (since it requires the config file in the first place)